lib/vagrant-service-manager/plugin_util.rb

```diff
@@ -117,6 +126,32 @@ def self.env_label(script_readable)
           'windows'
         end
       end
+
+      def self.download_url(machine, base_url, version)
+        os_type = if OS.windows?
+                    'Windows'
+                  elsif OS.unix?
+                    'Linux'
+                  elsif OS.mac?
+                    'Darwin'
+                  end
+
+        arch = machine.env.host.capability(:os_arch)
+        "#{base_url}#{os_type}/#{arch}/docker-#{version}"
+      end
+
+      def self.binary_downloaded?(version)
+        File.file?("#{BIN_FOLDER}docker-#{version}")
+      end
+
+      def self.download_binary(ui, url)
+        path = BIN_FOLDER + File.basename(url)
+
+        ui.info 'Downloading binary from url ' + url
+        Vagrant::Util::Downloader.new(url, path).download!
```

Comment:
Right now we discuss to place it to current folder. so that is is easy to call like
```
$ vagrant service-manager install-cli docker
$ ./docker-1.9.1 images
```

I like suggestion on adding extra option for path where user might want to place binary.